app/src/main/contrib/components/Responsibilities/BurdenEstimates/UploadBurdenEstimatesPage.tsx

```diff
@@ -1,29 +1,37 @@
 import * as React from "react";
-import { touchstoneActions } from "../../../actions/TouchstoneActions";
-import { responsibilityActions } from "../../../actions/ResponsibilityActions";
-import { responsibilityStore } from "../../../stores/ResponsibilityStore";
-import { modellingGroupActions } from "../../../../shared/actions/ModellingGroupActions";
-import { PageWithHeaderAndNav } from "../../PageWithHeader/PageWithHeaderAndNav";
-import { doNothing } from "../../../../shared/Helpers";
-import { DownloadDataTitle } from "../DownloadDataTitle";
-import { UploadBurdenEstimatesContent } from "./UploadBurdenEstimatesContent";
-import { estimateTokenActions } from "../../../actions/EstimateActions";
+import {touchstoneActions} from "../../../actions/TouchstoneActions";
+import {responsibilityActions} from "../../../actions/ResponsibilityActions";
+import {responsibilityStore} from "../../../stores/ResponsibilityStore";
+import {modellingGroupActions} from "../../../../shared/actions/ModellingGroupActions";
+import {PageWithHeaderAndNav} from "../../PageWithHeader/PageWithHeaderAndNav";
+import {doNothing} from "../../../../shared/Helpers";
+import {DownloadDataTitle} from "../DownloadDataTitle";
+import {UploadBurdenEstimatesContent} from "./UploadBurdenEstimatesContent";
+import {estimateTokenActions} from "../../../actions/EstimateActions";
+import {
+    processEncodedResultAndNotifyOnErrors
+} from "../../../../shared/sources/Source";
+import {queryStringAsObject} from "../../../../shared/Helpers";
 
-interface LocationProps {
+export interface UploadEstimatesProps {
     groupId: string;
     touchstoneId: string;
     scenarioId: string;
 }
 
-export class UploadBurdenEstimatesPage extends PageWithHeaderAndNav<LocationProps> {
+export class UploadBurdenEstimatesPage extends PageWithHeaderAndNav<UploadEstimatesProps> {
     componentDidMount() {
         setTimeout(() => {
+
+            processEncodedResultAndNotifyOnErrors<string>(queryStringAsObject());
+
             estimateTokenActions.clearUsedToken();
             modellingGroupActions.setCurrentGroup(this.props.location.params.groupId);
             responsibilityStore.fetchTouchstones().catch(doNothing).then(() => {
                 touchstoneActions.setCurrentTouchstone(this.props.location.params.touchstoneId);
                 responsibilityStore.fetchResponsibilities().catch(doNothing).then(() => {
                     responsibilityActions.setCurrentResponsibility(this.props.location.params.scenarioId);
+                    responsibilityActions.setRedirectPath(this.props.location.pathname);
                     responsibilityStore.fetchOneTimeEstimatesToken().catch(doNothing)
```

Comment:
It might be a good idea to not directly expose `fetchOneTimeEstimatesToken` as part of the public signature of `responsibilityStore`, and instead expose as wrapper method that takes an optional redirect path. Then, the wrapper method can take care of calling `setRedirectPath`, and calling code is less error prone.

For an example of wrapping a fetch method like this, see `OneTimeTokenStoreInterface._fetchToken()`.